pylib/Utilities/Logger.py

```diff
@@ -119,34 +125,37 @@ def log_to_elk(self, result, logtype):
         if self.elk_head is None or self.elk_id is None:
             self.verbose_print('Error: entered log_to_elk() function without elk_id and elk_head specified')
             return
-        if 'MTT_ELK_MAXSIZE' in os.environ:
-            try:
-                maxsize = int(os.environ['MTT_ELK_MAXSIZE'])
-            except:
-                maxsize = None
-            if maxsize is not None and 'stdout' in result and len(result['stdout']) > maxsize:
-                if maxsize > 0:
-                    result['stdout'] = ['<truncated>'] + result['stdout'][-maxsize:]
-                else:
-                    result['stdout'] = ['<truncated>']
-            if maxsize is not None and 'stderr' in result:
-                if maxsize > 0:
-                    result['stderr'] = ['<truncated>'] + result['stderr'][-maxsize:]
-                else:
-                    result['stderr'] = ['<truncated>']
-        if 'MTT_ELK_NOSTDOUT' in os.environ and 'stdout' in result:
-            result['stdout'] = ['<ignored>']
-        if 'MTT_ELK_NOSTDERR' in os.environ and 'stderr' in result:
-            result['stderr'] = ['<ignored>']
+
         result = {k:(str(v) if isinstance(v, datetime.datetime)
                             or isinstance(v, datetime.date) else v)
                   for k,v in result.items()}
         result['execid'] = self.elk_id
+        result['cycleid'] = self.elk_cycleid
+        result['caseid'] = self.elk_caseid
         result['logtype'] = logtype
+
+        # drop the stderr and stdout, will use the copies that are part of the commands issues
+        if 'stderr' in result:
+            del result['stderr']
```

Comment:
If MTT captures an exception, maybe due to an error in Python, that exception is stored in `result['stderr']`

The `stderr` and `stdout` in the commands array is only populated from ExecuteCmd plugin and doesn't include when a syntax error happens within another plugin in MTT.

Reply:
the observed behavior is that the last command issued is also populating this outer `stderr` and `stdout` which is why I dropped it

Reply:
I think with your pull request, we should merge it since it works, and then we can do some testing to see if all the errors we care about make it into ELK